### PR TITLE
Add sidebar_checker.sh

### PR DESCRIPTION
**Pull request by VadymSachenko**
## PR Description
This PR adds the sidebar checker script, which can detect documents that are absent in sidebars.
TBD

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: _scripts/sidebar_checker/run-sidebar-checker.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+1. In VS Code, open Terminal.
+2. Run the script:
+```bash
+bash _scripts/sidebar-checker/sidebar_checker.sh
+```
+
+This prints a message in the terminal indicating which files are missing and generates the `missing-documents.yml` file with missing documents. The file is saved in the `_scripts/sidebar_checker` folder.
+
+**Note:** If you don't need the `missing-documents.yml` file, remove it in order to not push it to GitHub.
```
Comment (andriitserkovnyi) [inline _scripts/sidebar_checker/run-sidebar-checker.md:25]: Add `missing-documents.yml` to `.gitignore` and remove this note. Also remove the note from the script.

File: _scripts/sidebar_checker/run-sidebar-checker.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+This document explains how Sidebar Checker works and how to run it.
+
+*Sidebar Checker* is a script that checks sidebars for missing documents in the docs folder and generates a YAML file with the missing files.
```
Comment (andriitserkovnyi) [inline _scripts/sidebar_checker/run-sidebar-checker.md:3]: This sentence could use more clarity. Try changing `documents` with `entries` and adjusting the rest of the sentence.

File: _scripts/sidebar_checker/run-sidebar-checker.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+3. Removes the output file if it exists.
+4. For each sidebar, the script looks for missing files in the corresponding directory.
+5. If missing files are found, the script prints a message indicating which files are missing and writes the missing files to a YAML file.
+
+## Run the script
+
+1. In VS Code, open Terminal.
+2. Run the script:
+```bash
+bash _scripts/sidebar-checker/sidebar_checker.sh
```
Comment (andriitserkovnyi) [inline _scripts/sidebar_checker/run-sidebar-checker.md:20]:
```suggestion
bash _scripts/sidebar_checker/sidebar_checker.sh
```

File: _scripts/sidebar_checker/sidebar_checker.sh [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# Define doc folders
+FOLDERS=("docs/acp/user" "docs/cloud/dev" "docs/marketplace/dev" "docs/marketplace/user" "docs/pbc/all" "docs/scos/dev" "docs/scos/user" "docs/scu/dev" "docs/sdk/dev")
+
+# Define sidebar file paths
+SIDEBARS=("_data/sidebars/acp_user_sidebar.yml" "_data/sidebars/cloud_dev_sidebar.yml" "_data/sidebars/marketplace_dev_sidebar.yml" "_data/sidebars/marketplace_user_sidebar.yml" "_data/sidebars/pbc_all_sidebar.yml" "_data/sidebars/scos_dev_sidebar.yml" "_data/sidebars/scos_user_sidebar.yml" "_data/sidebars/scu_dev_sidebar.yml" "_data/sidebars/sdk_dev_sidebar.yml")
+
+# Define sidebar titles
+TITLES=("ACP User" "Cloud Dev" "Marketplace Dev" "Marketplace User" "PBC All" "SCOS Dev" "SCOS User" "SCU Dev" "SDK Dev")
+
+# Define list of ignored folders
```
Comment (andriitserkovnyi) [inline _scripts/sidebar_checker/sidebar_checker.sh:18]:
```suggestion
# Define the folders to ignore
```

File: _scripts/sidebar_checker/sidebar_checker.sh [hunk 5]
```diff
@@ -0,0 +1,69 @@
+
+
+# To run the scipt, in Terminal, run `bash _scripts/sidebar_checker/check_all_sidebars.sh`.
+# The `missing_documents.yml` file with missing documents is generated and saved in the `_scripts/sidebar_checker` folder.
+# Note: If you don't need the `missing_documents.yml` file, remove it in order not to push it to GuitHub.
+
+#!/bin/bash
+
+# Define doc folders
```
Comment (andriitserkovnyi) [inline _scripts/sidebar_checker/sidebar_checker.sh:9]:
```suggestion
# Define doc folder paths
```

**Comment by andriitserkovnyi** [comment]
Great job on this one

File: _scripts/sidebar_checker/run-sidebar-checker.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+*Sidebar Checker* is a script that checks sidebars for missing documents in the docs folder and generates a YAML file with the missing files.
+
+## Overview of Sidebar Checker
+
+The script does the following: 
+
+1. Defines the directories to check, the sidebars to update, and the sidebar titles.
+2. Defines a list of folders to ignore.
+3. Removes the output file if it exists.
```
Comment (VadymSachenko) [inline _scripts/sidebar_checker/run-sidebar-checker.md:11]:
```suggestion
3. Removes the YAML output file if it exists.
```

File: _scripts/sidebar_checker/run-sidebar-checker.md (hunk 2 above)
Comment (VadymSachenko) [reply _scripts/sidebar_checker/run-sidebar-checker.md:3]:
```suggestion
*Sidebar Checker* is a script that checks sidebars for missing entries in the `docs` folder and generates a YAML file with these missing entries.
```

File: _scripts/sidebar_checker/sidebar_checker.sh [hunk 7]
```diff
@@ -0,0 +1,69 @@
+
+
+# To run the scipt, in Terminal, run `bash _scripts/sidebar_checker/check_all_sidebars.sh`.
+# The `missing_documents.yml` file with missing documents is generated and saved in the `_scripts/sidebar_checker` folder.
+# Note: If you don't need the `missing_documents.yml` file, remove it in order not to push it to GuitHub.
```
Comment (VadymSachenko) [inline _scripts/sidebar_checker/sidebar_checker.sh:5]:
```suggestion
```

File: _scripts/sidebar_checker/run-sidebar-checker.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+This document explains how Sidebar Checker works and how to run it.
+
+*Sidebar Checker* is a script that checks sidebars for missing entries in the `docs` folder and generates a YAML file with these missing entries.
```
Comment (andriitserkovnyi) [inline _scripts/sidebar_checker/run-sidebar-checker.md:3]:
```suggestion
*Sidebar Checker* is a script that generates a YAML report of missing sidebar entries based on the documents in the `docs` folder.
```

**Pull request merged**